### PR TITLE
Fix numpy warnings

### PR DESCRIPTION
**Pull request by waltsims**

### CODE REVIEW

**Comment by waltsims** [comment]
Matlab_mask is the current issue. I have to replicate legacy behavior.

**Comment by waltsims** [comment]
There is a lot of weirdness in here with the uint8 typing. We should clean this up in the future. For thow this seems to allow the CI to pass.

File: kwave/utils/matlab.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ def matlab_find(arr: Union[List[int], np.ndarray], val: int = 0, mode: str = "ne
     return np.expand_dims(arr, -1)  # compatibility, n => [n, 1]
 
 
+@typechecker
```
Comment (waltsims) [inline kwave/utils/matlab.py:78]: This change is not required for this PR but I just added it while I was in the file

File: kwave/utils/matlab.py [hunk 2]
```diff
@@ -89,6 +90,9 @@ def matlab_mask(arr: np.ndarray, mask: np.ndarray, diff: Optional[int] = None) -
 
     """
 
+    if mask.dtype == "uint8":
+        mask = mask.astype(np.int16)
+
     if diff is None:
         return np.expand_dims(arr.ravel(order="F")[mask.ravel(order="F")], axis=-1)  # compatibility, n => [n, 1]
     else:
```
Comment (waltsims) [inline kwave/utils/matlab.py:98]: I think there was an issue with unit8 roll-over to positive indexes. This is solved with negative indexing in python and therefore using int16 is acceptable.

File: kwave/utils/kwave_array.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 import numpy as np
 from numpy import arcsin, pi, cos, size, array
-from numpy.linalg import linalg
```
Comment (waltsims) [inline kwave/utils/kwave_array.py:9]: numpy depricated this import structure for the new one I modified

File: kwave/ktransducer.py [hunk 4]
```diff
@@ -644,7 +644,7 @@ def delay_mask(self, mode=None):
             ].min()  # -1s compatibility
         else:
             mask[unflatten_matlab_mask(mask, active_elements_index - 1)] += self.stored_beamforming_delays_offset  # -1s compatibility
-        return mask.astype(np.uint8)
```
Comment (waltsims) [inline kwave/ktransducer.py:647]: Again here for one of the operations, the unit8 type was not being correctly promoted for values larger than 255 so I bumped the type.

File: kwave/utils/matlab.py [hunk 5]
```diff
@@ -89,6 +90,9 @@ def matlab_mask(arr: np.ndarray, mask: np.ndarray, diff: Optional[int] = None) -
 
     """
 
+    if mask.dtype == "uint8":
```
Comment (faridyagubbayli) [inline kwave/utils/matlab.py:93]: what if we convert uint8 to uint16 and int8 to int16?

**Review by faridyagubbayli — APPROVED** [review]
LGTM! Left a comment

File: kwave/utils/matlab.py (hunk 5 above)
Comment (waltsims) [reply kwave/utils/matlab.py:93]: Exaclty, current "value -1 is out of bounds of uint8 type"

Comment (waltsims) [reply kwave/utils/matlab.py:93]:
```python
>>> x
array([0], dtype=uint8)
>>> x - 1
array([255], dtype=uint8)
>>> np.version
<module 'numpy.version' from '/Users/walters/miniconda3/lib/python3.12/site-packages/numpy/version.py'>
>>> np.__version__
'1.26.4'
```

Here is the legacy behavior

Comment (waltsims) [reply kwave/utils/matlab.py:93]: Here is the new behavior:

```python
>>> import numpy as np
>>> x = np.array([0], dtype='uint8')
>>> x - 1
array([255], dtype=uint8)
>>> np.__version__
'2.1.3'

```

looks like my assumption was wrong

Comment (waltsims) [reply kwave/utils/matlab.py]: Seems to be an issue with type promotion with the + operator.

```python

>>> import numpy as np
>>> x = np.array([0], dtype='uint8')
>>> x - 1
array([255], dtype=uint8)
>>> x.ravel(order='F') + -1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: Python integer -1 out of bounds for uint8
>>> x.ravel(order='F').astype("int8") + -1
array([-1], dtype=int8)
>>> x + -1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: Python integer -1 out of bounds for uint8
>>> np.__version__
'2.1.3'
```

Comment (waltsims) [reply kwave/utils/matlab.py:93]: This is all related to this:
https://github.com/numpy/numpy/issues/25639#issuecomment-1901166868

**Pull request merged**